api/dataprocessor.go

```diff
@@ -466,27 +465,25 @@ func (s *Server) getSeriesCachedStore(ctx *requestContext, until uint32) []chunk
 
 	// the request cannot completely be served from cache, it will require cassandra involvement
 	if !cacheRes.Complete {
-		storeIterGens := make([]chunk.IterGen, 0)
-
 		if cacheRes.From != cacheRes.Until {
-			storeIterGens, err = s.BackendStore.Search(key, cacheRes.From, cacheRes.Until)
+			storeIterGens, err := s.BackendStore.Search(key, cacheRes.From, cacheRes.Until)
 			if err != nil {
 				panic(err)
 			}
-		}
 
-		for _, itgen := range storeIterGens {
-			it, err := itgen.Get()
-			if err != nil {
-				// TODO(replay) figure out what to do if one piece is corrupt
-				log.Error(3, "itergen: error getting iter from cassandra slice %+v", err)
-				continue
+			for _, itgen := range storeIterGens {
+				it, err := itgen.Get()
+				if err != nil {
+					// TODO(replay) figure out what to do if one piece is corrupt
+					log.Error(3, "itergen: error getting iter from cassandra slice %+v", err)
+					continue
+				}
+				// it's important that the itgens get added in chronological order,
+				// currently we rely on cassandra returning results in order
+				go s.Cache.Add(key, prevts, itgen)
+				prevts = itgen.Ts()
+				iters = append(iters, *it)
 			}
-			// it's important that the itgens get added in chronological order,
-			// currently we rely on cassandra returning results in order
-			go s.Cache.Add(key, prevts, itgen)
-			prevts = itgen.Ts()
-			iters = append(iters, *it)
```

Comment:
this commit doesn't seem to have anything to do with what it claims to be about.
the actual fix somehow snuck in " figure out previous chunk on cache add" and this change is functionality identical than the previous code.

protip: use `git add -p` to add the right stuff to the right commit, and `git rebase -i` to make adjustments if needed.
in a few minutes you can set up the history in such a way that I can save many more minutes during the review process, and you'll also save minutes to anyone in the future trying to make sense of what happened to this code.